samplesheets/assayapps/generic/plugins.py

```diff
@@ -0,0 +1,210 @@
+            col_value = self._get_col_value(column_name, row, table)
+            if col_value:
+                data_collections.append(col_value)
+
+        # Build iRODS path from list and stop at first None value
+        if data_collections:
+            data_path = '/' + '/'.join(data_collections)
+            return assay_path + data_path
+        return None
+
+    def update_row(self, row, table, assay, index):
+        """
+        Update render table row with e.g. links. Return the modified row.
+
+        :param row: Original row (list of dicts)
+        :param table: Full table (dict)
+        :param assay: Assay object
+        :param index: Row index (int)
+        :return: List of dicts
+        """
+        if not settings.IRODS_WEBDAV_ENABLED or not assay:
+            return row
+        assay_path = self.get_assay_path(assay)
+        if not assay_path:
+            return row
+
+        base_url = settings.IRODS_WEBDAV_URL + assay_path
+        top_header = None
+        th_colspan = 0
+
+        results_cols = assay.comments.get(RESULTS_COMMENT)
+        if results_cols:
+            results_cols = results_cols.lower().split(';')
+        misc_cols = assay.comments.get(MISC_FILES_COMMENT)
+        if misc_cols:
+            misc_cols = misc_cols.lower().split(';')
+        data_cols = assay.comments.get(DATA_LINK_COMMENT)
+        if data_cols:
+            data_cols = data_cols.lower().split(';')
+            if table['irods_paths'][index]:
+                row_path = table['irods_paths'][index]['path']
+            else:
+                row_path = self.get_row_path(row, table, assay, assay_path)
+
+        for i in range(len(row)):
+            header = table['field_header'][i]
+            if not top_header or i >= th_colspan:
+                top_header = get_top_header(table, i)
+                th_colspan += top_header['colspan']
+
+            # TODO: Check if two comments reference the same column header?
+            # Create Results links
+            if results_cols:
+                if self._link_from_comment(
+                    row[i],
+                    header,
+                    top_header,
+                    results_cols,
+                    f'{base_url}/{RESULTS_COLL}',
+                ):
+                    continue
+            # Create MiscFiles links
+            if misc_cols:
+                if self._link_from_comment(
+                    row[i],
+                    header,
+                    top_header,
+                    misc_cols,
+                    f'{base_url}/{MISC_FILES_COLL}',
+                ):
+                    continue
+            # Create DataCollection links
+            if data_cols:
+                if self._link_from_comment(
+                    row[i],
+                    header,
+                    top_header,
+                    data_cols,
+                    f'{settings.IRODS_WEBDAV_URL}{row_path}',
+                ):
+                    continue
```

Comment:
The last `if` clause and `continue` can be omitted, as they are at the end of the `for` loop and we go back to the beginning here anyway.

Reply:
True, but I wanted to keep the three blocks similar.